components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java

```diff
@@ -1103,4 +1113,21 @@ private boolean isFederatedUser(OAuthTokenReqMessageContext tokReqMsgCtx) {
         }
         return tokReqMsgCtx.getAuthorizedUser().isFederatedUser();
     }
+
+
+    private void setACRValueInTheToken(OAuthTokenReqMessageContext tokenReqMsgCtxt, AccessTokenDO accessTokenDO) {
+
+        Optional<String> optionalSelectedACR = OAuth2Util.getSelectedACRValue(tokenReqMsgCtxt);
+        if (!optionalSelectedACR.isPresent()) {
+            if (log.isDebugEnabled()) {
+                log.debug("ACR value is not available for the token issued for '" +
+                        tokenReqMsgCtxt.getOauth2AccessTokenReqDTO().getClientId() + "' client.");
+            }
+            return;
+        }
+        String selectedACRValue = optionalSelectedACR.get();
+        accessTokenDO.setAcr(selectedACRValue);
+        tokenReqMsgCtxt.setAcr(selectedACRValue);
```

Comment:
Check if we really need that or unify these.

Reply:
Here we don't need to set the ACR value to the `tokenReqMsgCtxt` object. The usage of the setter and getter will only be required in the RefreshGrantHandler class. Check if we really need this attribute in the TokenRequestContext class (If we have any other way to get the ACR value to the RefreshTokenGrantHandler, then we can remove this attribute from tokenReqMsgCtxt class)